src/routes/event/add.tsx

```diff
@@ -18,39 +19,88 @@ export default function EventAdd() {
     });
   }, []);
   const [location, setLocation] = React.useState<string | null>(null);
+  const [category, setCategory] = React.useState<string | null>(null);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const data = new FormData(event.currentTarget);
+    const eventName = data.get("eventName") as string;
+    const eventDescription = data.get("eventDescription") as string;
+    const eventStartDate = data.get("eventStartDate") as string;
+    const eventEndDate = data.get("eventEndDate") as string;
+    const eventLocation = data.get("eventLocation") as string;
+    const eventCategory = data.get("eventCategory") as string;
+    const newEventKey = push(ref(database, "events")).key;
+    if (newEventKey === null) {
+      return;
+    }
+    const eventNameData = {
+      [newEventKey]: eventName,
+    };
+    const eventDetailsData = {
+      eventDescription: eventDescription,
+      eventOrganizer: auth.currentUser?.uid,
+      eventStartDate: eventStartDate,
+      eventEndDate: eventEndDate,
+      eventLocation: eventLocation,
+      eventCategory: eventCategory,
+    };
+    const categoryData = {
+      [newEventKey]: true,
+    };
+    const locationData = {
+      [newEventKey]: true,
+    };
+    set(ref(database, "event-details/" + newEventKey), eventDetailsData);
+    update(ref(database, "event-names"), eventNameData);
+    update(ref(database, "locations/" + eventLocation), locationData);
+    if (eventCategory !== null && eventCategory !== "") {
+      update(ref(database, "categories/" + eventCategory), categoryData);
+    }
+    navigate("/event/list");
```

Comment:
you have this being called twice
you can extract it to a function and reuse it
since navigation happens a lot, we normally create a separate util file, that holds your "redirects" in 1 place, and you just reuse them throughout the project.